### PR TITLE
Add Dockerfile and modify official installation to support ruby api

### PR DESCRIPTION
**Pull request by rtfpessoa**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,6 @@
 db/*.sqlite3
 db/jwt-rsa.key
 vendor/
+.idea/
+*.iml
+.vscode
```
Comment (jcs) [inline .gitignore]: Are these relevant to Docker?

File: scripts/bitwarden.sh [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+| |__ (_) |___      ____ _ _ __ __| | ___ _ __
+| '_ \| | __\ \ /\ / / _` | '__/ _` |/ _ \ '_ \
+| |_) | | |_ \ V  V / (_| | | | (_| |  __/ | | |
+|_.__/|_|\__| \_/\_/ \__,_|_|  \__,_|\___|_| |_|
+
+EOF
+
+cat << EOF
+Open source password management solutions
+Copyright 2015-$(date +'%Y'), 8bit Solutions LLC
```
Comment (jcs) [inline scripts/bitwarden.sh]: If this file came from the upstream Bitwarden repos, then this would probably come with an (A)GPL license which is not acceptable for this project.

File: scripts/install.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
```
Comment (jcs) [inline scripts/install.sh]: I'm not familiar with Docker but do these files need to be named `script/install.sh` and `script/run.sh`?  I would prefer they be put in a directory just for Docker things so it's clear what they're for.  Also, these will need license and copyright headers like the other files.

File: scripts/run.sh [hunk 4]
```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
```
Comment (jcs) [inline scripts/run.sh]: Same here.

File: tools/setup.rb [hunk 5]
```diff
@@ -0,0 +1,536 @@
+#!/usr/bin/env ruby
```
Comment (jcs) [inline tools/setup.rb]: Same here, I'd prefer this be moved into a `docker` directory so it's clear what it's for.  `setup.rb` is kind of vague.

**Comment by jcs** [comment]
Also, if you want to add a blurb to `README.md` to quickly explain how to install this through Docker, that'd be appreciated.

File: .gitignore (hunk 1 above)
Comment (rtfpessoa) [reply .gitignore]: These are just IDE files to be ignored. If you mind I can remove them, but I guess they do not cause any problem.

**Comment by rtfpessoa** [comment]
@jcs code updated with the changes.

**Comment by jcs** [comment]
The files from upstream with an AGPL license are not acceptable for this project, so you would need to rewrite those from scratch and attach your own free license.

**Comment by rtfpessoa** [comment]
I really should learn more about licenses. Rewriting those files seems like a very weird thing since they do close to nothing. But I will try, but it might take a bit more.

**Comment by mbirth** [comment]
Are the upstream files actually needed for a run-of-the-mill fire-and-forget Docker image?

**Comment by jcs** [comment]
Please host the instructions somewhere and submit a PR to link to it from the Deployment section of README.md.  I'd rather not have to maintain documentation for all kinds of different deployments.
